### PR TITLE
Add mkldnn mul operator

### PR DESCRIPTION
**Pull request by XiaobingSuper**
### mkldnn backward ops list:
 - [ ] \(https://github.com/pytorch/pytorch/pull/20567) Add aten mkldnn conv2d backward operator :yellow_heart:
 - [ ] \(https://github.com/pytorch/pytorch/pull/20570) Add aten mkldnn backward ops: relu, linear and reshape :yellow_heart:
 - [ ] \(https://github.com/pytorch/pytorch/pull/20571) Add aten mkldnn backward ops: max_pool2d, avg_pool2d and adaptive_avg_poo2d :yellow_heart:
 - [ ] \(https://github.com/pytorch/pytorch/pull/20572) Add aten mkldnn batchnorm backward operator :yellow_heart:
 - [ ] \(https://github.com/pytorch/pytorch/pull/20573) Add aten mkldnn zero_ operator:yellow_heart:
 - [ ] \(https://github.com/pytorch/pytorch/pull/20575) Add mkldnn mul operator :💚

### CODE REVIEW

**Comment by XiaobingSuper** [comment]
This PR depend on https://github.com/pytorch/pytorch/pull/20569, which adds a new ideep mul API. Thanks!

**Comment by XiaobingSuper** [comment]
Please first solve the issue https://github.com/pytorch/pytorch/issues/20895 problem, thanks!

**Comment by bddppq** [comment]
#20569 has been merged to master, please rebase

**Comment by XiaobingSuper** [comment]
@bddppq, rebased, please help review it, thanks!

**Comment by XiaobingSuper** [comment]
@mingfeima, @uyongw, @Jianhui-Li, @jgong5

File: aten/src/ATen/native/mkldnn/BinaryOps.cpp [hunk 1]
```diff
@@ -63,6 +75,38 @@ Tensor& mkldnn_add_(Tensor& self, const Tensor& other, Scalar alpha) {
   return native::mkldnn_add_out(self, self, other, alpha);
 }
 
+Tensor& mkldnn_mul_out(Tensor& result, const Tensor& self, const Tensor& other) {
+  AT_ASSERTM(result.sizes() == self.sizes(),
+             "mkldnn_mul_out: currently mkldnn not support broadcasting");
```
Comment (bddppq) [inline aten/src/ATen/native/mkldnn/BinaryOps.cpp]: If it's about broadcasting, you probably should check `self.sizes() == other.sizes()`?

Comment (XiaobingSuper) [reply aten/src/ATen/native/mkldnn/BinaryOps.cpp]: Perharps the warning message should be "mkldnn_mul_out: the output size should be same as input size"? I want to make sure the output size should be same as input, the broadcasting checking in Line-92. Thanks!

Comment (bddppq) [reply aten/src/ATen/native/mkldnn/BinaryOps.cpp]: Yeah makes senses

File: test/test_mkldnn.py [hunk 2]
```diff
@@ -216,6 +216,55 @@ def test_add(self):
         torch.add(mx, my, alpha=alpha, out=mkldnn_out)
         self.assertEqual(out, mkldnn_out.to_dense())
 
+    def test_mul(self):
+        N = torch.randint(3, 10, (1,)).item()
```
Comment (VitalyFedyunin) [inline test/test_mkldnn.py]: Please do not relay on random numbers.  You are not freezing random seed, that means every test execution will generate new shape. It might create false sense of security, but many modern test frameworks will attempt to rerun failed test hiding problem prom developers.

Much better if you explicitly specify shapes you are interested to test.

File: test/test_mkldnn.py [hunk 3]
```diff
@@ -216,6 +216,55 @@ def test_add(self):
...
+        C = torch.randint(3, 100, (1,)).item()
+        value = torch.randn(1, dtype=torch.float32).item()
```
Comment (VitalyFedyunin) [inline test/test_mkldnn.py]: `value = 1.`

**Review by VitalyFedyunin — COMMENTED** [review]
Overall looks good, but need some tests improvement.
